front/app/containers/LandingPage/SignedOutHeader/HeaderContent.tsx

```diff
@@ -0,0 +1,211 @@
+import useAppConfiguration from 'hooks/useAppConfiguration';
+import { isNilOrError } from 'utils/helperUtils';
+import { trackEventByName } from 'utils/analytics';
+import tracks from '../tracks';
+import { openSignUpInModal } from 'components/SignUpIn/events';
+import { InjectedIntlProps } from 'react-intl';
+import messages from '../messages';
+import { injectIntl } from 'utils/cl-intl';
+
+const Container = styled.div<{
+  align: 'center' | 'left';
+  alignTo: 'center' | 'flex-start' | undefined;
+}>`
+  height: 100%;
+  max-width: ${({ theme }) => theme.maxPageWidth + 60}px;
+  padding: ${({ align }) => (align === 'left' ? '50px 100px' : '50px 30px')};
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: ${({ alignTo }) => alignTo || 'normal'};
+  z-index: 1;
+  box-sizing: content-box;
+  overflow-wrap: break-word;
+  word-wrap: break-word;
+  word-break: break-word;
+
+  ${media.smallerThanMaxTablet`
+    padding: '50px 30px';
+  `}
+`;
+
+const HeaderTitle = styled.h1<{
+  hasHeader: boolean;
+  fontColors: 'light' | 'dark';
+  align: 'center' | 'left';
+}>`
+  width: 100%;
+  color: ${({ hasHeader, fontColors, theme }) =>
+    hasHeader
+      ? fontColors === 'light'
+        ? '#fff'
+        : theme.colorMain
+      : theme.colorMain};
+  font-size: ${({ theme }) =>
+    theme.signedOutHeaderTitleFontSize || fontSizes.xxxl}px;
+  font-weight: ${({ theme }) => theme.signedOutHeaderTitleFontWeight || 600};
+  line-height: normal;
+  text-align: ${({ align }) => align};
+  padding: 0;
+  margin-bottom: 10px;
+
+  ${media.smallerThanMaxTablet`
+  font-size: ${fontSizes.xxxl}px;
+  `}
+
+  ${media.smallerThanMinTablet`
+    margin-bottom: 15px;
+  `}
+`;
+
+const HeaderSubtitle = styled.h2<{
+  hasHeader: boolean;
+  fontColors: 'light' | 'dark';
+  align: 'center' | 'left';
+  displayHeaderAvatars: boolean;
+}>`
+  width: 100%;
+  color: ${({ hasHeader, fontColors, theme }) =>
+    hasHeader
+      ? fontColors === 'light'
+        ? '#fff'
+        : theme.colorMain
+      : theme.colorMain};
+  font-size: ${fontSizes.large}px;
+  line-height: 28px;
+  font-weight: 400;
+  text-align: ${({ align }) => align};
+  text-decoration: none;
+  overflow-wrap: break-word;
+  word-wrap: break-word;
+  word-break: break-word;
+  padding: 0;
+  margin: 0;
+  margin-bottom: 15px;
+
+  ${({ displayHeaderAvatars }) =>
+    // needed because we don't always
+    // show avatars
+    !displayHeaderAvatars &&
+    `
+      margin-bottom: 38px;
+
+      ${media.smallerThanMinTablet`
+        margin-bottom: 30px;
+      `}
+  `}
+`;
+
+const StyledAvatarBubbles = styled(AvatarBubbles)`
+  min-height: 40px;
+  margin-bottom: 30px;
+
+  ${media.smallerThanMinTablet`
+    margin-bottom: 30px;
+  `}
+`;
+
+const SignUpButton = styled(Button)``;
+
+type TAlign = 'center' | 'left';
+interface Props {
+  fontColors: 'light' | 'dark';
+  align?: TAlign;
+}
+
+function getButtonStyle(fontColors: 'light' | 'dark') {
```

Comment:
What about something like:

```
type FontColors = 'light' | 'dark'
const BUTTON_STYLE_MAP: Record<FontColors, string> = { light: 'primary-inverse', dark: 'primary' }
const getButtonStyle = (fontColors: FontColors) => BUTTON_STYLE_MAP[fontColors];
```

Since `fontColors` is not an optional parameter and both possible values return something, why add `undefined` as an extra possible return type? Same for `getAlignItems`

Reply:
👍 feel free to add later!